core/src/main/scala/kafka/server/FinalizedFeatureChangeListener.scala

```diff
@@ -160,10 +160,12 @@ class FinalizedFeatureChangeListener(private val finalizedFeatureCache: ZkMetada
           // safe to ignore the exception if the thread is being shutdown. We raise the exception
           // here again, because, it is ignored by ShutdownableThread if it is shutting down.
           throw ie
-        case e: Exception => {
-          error("Failed to process feature ZK node change event. The broker will eventually exit.", e)
+        case cacheUpdateException: FeatureCacheUpdateException =>
```

Comment:
Thank you for your patch! I'm not a reviewer but I checked because this can be a problem in our own environment, too.

If `IllegalStateException` is thrown because an unknown `FeatureZNodeStatus` is found in the zk node or double invocations when a non-empty notifier was provided in the constructor, is it unnecessary to shut the broker down? 
Or, how about outputting an error like `IllegalArgumentException` handling inside `updateLatestOrThrow`?

Reply:
It's tough to decide what to do in the case where we hit a feature notification and we can't understand it. This is something we can discuss in a follow-on PR.

Reply:
I see. I totally agree that Zookeeper session expiry should be addressed fast, so I will create a follow-on PR if necessary.